codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -1143,85 +831,43 @@ public long getTimeMillisAndRemove(T name, long defaultValue) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public boolean equals(Object o) {
         if (!(o instanceof DefaultHeaders)) {
             return false;
         }
-
-        @SuppressWarnings("unchecked")
-        DefaultHeaders<T> h2 = (DefaultHeaders<T>) o;
-        // First, check that the set of names match. Don't use a TreeSet for comparison
-        // because we want to force the keyComparator to be used for all comparisons
-        List<T> namesList = namesList();
-        List<T> otherNamesList = h2.namesList();
-        if (!equals(namesList, otherNamesList, keyComparator)) {
-            return false;
-        }
-
-        // Compare the values for each name. Don't use a TreeSet for comparison
-        // because we want to force the valueComparator to be used for all comparisons
-        Set<T> names = new TreeSet<T>(keyComparator);
-        names.addAll(namesList);
-        for (T name : names) {
-            if (!equals(getAll(name), h2.getAll(name), valueComparator)) {
-                return false;
-            }
-        }
-
-        return true;
+        DefaultHeaders<T> other = (DefaultHeaders<T>) o;
+        return size() == other.size() && map.equals(other.map);
```

Comment:
Do we have a unit test which exercises this method with two "equivalent" HttpHeader objects...but the cases are different between header names?